monoloco/visuals/webcam.py

```diff
@@ -0,0 +1,204 @@
+    args.checkpoint = dic_models['keypoints']
+
+    logger.configure(args, LOG)  # logger first
+
+    if args.output_types is None:
+        args.output_types = ['multi']
+
+    assert 'bird' not in args.output_types
+    if 'json' not in args.output_types:
+        assert len(args.output_types) == 1
+    else:
+        assert len(args.output_types) < 3
+
+    # Devices
+    args.device = torch.device('cpu')
+    args.pin_memory = False
+    if torch.cuda.is_available():
+        args.device = torch.device('cuda')
+        args.pin_memory = True
+    LOG.debug('neural network device: %s', args.device)
+
+    # Add visualization defaults
+    args.figure_width = 10
+    args.dpi_factor = 1.0
+
+    args.z_max = 10
+    args.show_all = True
+    args.no_save = True
+    args.batch_size = 1
+
+    # Make default pifpaf argument
+    args.force_complete_pose = True
+    LOG.info("Force complete pose is active")
+
+    # Configure
+    decoder.configure(args)
+    network.Factory.configure(args)
+    show.configure(args)
+    visualizer.configure(args)
+
+    return args, dic_models
+
+
+def webcam(args):
+
+    assert args.mode in 'mono'
+
+    args, dic_models = factory_from_args(args)
+
+    # Load Models
+    net = Loco(model=dic_models[args.mode], mode=args.mode, device=args.device,
+               n_dropout=args.n_dropout, p_dropout=args.dropout)
+
+    processor, pifpaf_model = processor_factory(args)
+    preprocess = preprocess_factory(args)
+
+    # Start recording
+    cam = cv2.VideoCapture(0)
+    visualizer_mono = None
+
+    while True:
+        start = time.time()
+        ret, frame = cam.read()
+        image = cv2.resize(frame, None, fx=args.scale, fy=args.scale)
+        height, width, _ = image.shape
+        print('resized image size: {}'.format(image.shape))
+        image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+        pil_image = Image.fromarray(image)
+
+        data = datasets.PilImageList(
+            [pil_image], preprocess=preprocess)
+
+        data_loader = torch.utils.data.DataLoader(
+            data, batch_size=1, shuffle=False,
+            pin_memory=False, collate_fn=datasets.collate_images_anns_meta)
+
+        for (image_tensors_batch, _, meta_batch) in data_loader:
+            pred_batch = processor.batch(
+                pifpaf_model, image_tensors_batch, device=args.device)
+
+            for idx, (pred, meta) in enumerate(zip(pred_batch, meta_batch)):
+                pred = [ann.inverse_transform(meta) for ann in pred]
+
+                if idx == 0:
+                    pifpaf_outs = {
+                        'pred': pred,
+                        'left': [ann.json_data() for ann in pred],
+                        'image': image}
+
+        if not ret:
+            break
+        key = cv2.waitKey(1)
+        if key % 256 == 27:
+            # ESC pressed
+            print("Escape hit, closing...")
+            break
+
+        intrinsic_size = [xx * 1.3 for xx in pil_image.size]
```

Comment:
This was adapted for the old version? Can you please check it what happens if we remove it? Thanks!

Reply:
It didn't seem to make any difference so I took it out